docs/software/commands.md

```diff
@@ -298,10 +298,13 @@ format.
         ledger<br>
 
 * **surveytopology**
-  `surveytopology?duration=DURATION&node=NODE_ID`<br>
+  `surveytopology?duration=DURATION&node=NODE_ID&surveyversion=0`<br>
```

Comment:
I don't think it should work like this:
relaying nodes don't have access to the encrypted survey result (ie: `SurveyResponseBody` is not visible on the wire), it's probably better to just change the surveying code to respond with the new extended result, and leave everything else as is.

That way:
* existing nodes will be able to relay requests and (encrypted) result even if they're running an older version of the survey code
* we can deprecate right away responding with "old style" survey responses
* old nodes can't survey the network (because they may receive responses from newer nodes), but that seems fine

Reply:
Hmm. My thinking was that as per our usual overlay processes, we provide backward compatibility for at least some time, which should allow old nodes to use survey functionality. We've never upgraded survey before, so perhaps we're lacking a process and backward compatibility expectations here. Given that this isn't threatening to consensus and can be mitigated with a version upgrade, maybe it's ok to stop supporting survey for older nodes (I don't know who is using though and if it will cause problems).

Also - the fact that old nodes can still properly process newer requests/responses is true for this particular version but it's not guaranteed in the future. As the survey format changes, we'll still have to do what this PR does.

Reply:
> Hmm. My thinking was that as per our usual overlay processes, we provide backward compatibility for at least some time, which should allow old nodes to use survey functionality. We've never upgraded survey before, so perhaps we're lacking a process and backward compatibility expectations here. Given that this isn't threatening to consensus and can be mitigated with a version upgrade, maybe it's ok to stop supporting survey for older nodes (I don't know who is using though and if it will cause problems).

I think it's acceptable to require surveying nodes to run the latest version.

> Also - the fact that old nodes can still properly process newer requests/responses is true for this particular version but it's not guaranteed in the future. As the survey format changes, we'll still have to do what this PR does.

I don't think this is true: the survey request and relaying messages should not change format. The only thing that should change is the format of the survey response (in response to a generic `SURVEY_REQUEST` message) and is encoded as an encrypted opaque blob.

Reply:
>I don't think this is true: the survey request and relaying messages should not change format. The only thing that should change is the format of the survey response (in response to a generic SURVEY_REQUEST message) and is encoded as an encrypted opaque blob.

I don't think we can guarantee this. We don't know what the survey needs will look like in the future. The request itself contains SurveyMessageCommandType which assumes the possibility of running survey types. All I'm trying to say is that it's not a problem right now, but we might still need to deal with this in the future